crates/rattler_repodata_gateway/src/gateway/local_subdir.rs

```diff
@@ -64,4 +64,9 @@ impl SubdirClient for LocalSubdirClient {
         })
         .await
     }
+
+    fn package_names(&self) -> Arc<[String]> {
```

Comment:
You are currently not caching this method so there is not need to return an `Arc`. Just use `Vec<String>`.